lib/diagnostic.js

```diff
@@ -6,26 +6,40 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizedWords = function getNormalizedWords(stringInput) {
+  let words = stringInput.split(/\s+/);
+  for(let i=0; i<words.length; i++){
+    words[i] = words[i].replace(/\W+/, '').toUpperCase();
+  }
+  return words;
 };
 
+
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function normalizedWords(stringInput) {
+  let uniqueWords = {};
+  let normalizedWordsList = normalizedWords(stringInput);
+  for(let i = 0; i < normalizedWordsList.length; i++){
+    uniqueWords[normalizedWordsList[i]] = 1;
+  }
+  return Object.keys(uniqueWords);
 };
 
 // 3. Write a function that returns the count of words in a string.
 // Provid the *option* to count unique words instead of total words.
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function wordCount(stringInput, unique) {
+  if(unique == false){
+    return normalizedWords(stringInput).length;
+  }else{
+    return normalizedWords(stringInput).length;
+  }
 };
 
```

Comment:
Almost, but my guess is you didn't check this one out. It calls the same function regardless.